### DISCUSSION TITLE
bean lookup disabled observer still working

### DISCUSSION THREAD

**Discussion by alex-martel** [quarkusio/quarkus#38349] [Q&A] [answered]
I got a bit surprised by a behaviour in quarkus, that I think is by design but I would like to confirm as may also be a bug.

I have a service bean that is `applicationscoped` and is runtime disabled via `@LookupIfProperty`. That service contains a observer method; `void myObserver(@Observes TheEvent event)`. I was surprised to see the class being constructed and the observer method to be called right after the event is fired from somewhere else. After that the bean is still non "resolvable", but I am surprised that it still call my observer.

I was under the impression I could have multiple services with observers, and only the ones with the lookup enabled would actually handle the events but looks like its not.

In any case I have a small reproducer here; https://github.com/manofthepeace/lookup-observer

**Comment by alex-martel** [comment]
@mkouba , sorry for the ping.

**Reply by mkouba** [reply, marked as answer]
It really works as designed. As explained in [the docs](https://quarkus.io/guides/cdi-reference#declaratively-choose-beans-that-can-be-obtained-by-programmatic-lookup) the `@LookupIfProperty` can only be used to narrow down the set of beans that can be obtained by programmatic lookup via `@Inject Instance<>`. It's very similar to how CDI `@Alternative` and observers work. You can't inject the bean that is "hidden" by an `@Alternative` bean but the observers are still notified.

**Reply by alex-martel** [reply]
Thanks a lot, that explains.

**Reply by mkouba** [reply]
yw ;-)
